src/node-explorer-provider.ts

```diff
@@ -0,0 +1,256 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import * as vscode from 'vscode';
+import * as path from 'path';
+import { FileInfo, Peer, Status } from './types';
+import { Tailscale } from './tailscale/cli';
+import { TSObjFileSystemProvider } from './tsobj-file-system-provider';
+
+export class NodeExplorerProvider
+  implements
+    vscode.TreeDataProvider<PeerBaseTreeItem>,
+    vscode.TreeDragAndDropController<PeerBaseTreeItem>,
+    vscode.FileDecorationProvider
+{
+  dropMimeTypes = ['text/uri-list']; // add 'application/vnd.code.tree.testViewDragAndDrop' when we have file explorer
```

Comment:
I think this PR is missing this commit: https://github.com/tailscale-dev/vscode-tailscale/commit/60dc7fa621a093511544cf41f8d2173da89511c2

At least the TypeScript part (not the Go part, since we're just shelling out to SSH) 

This way we can have scp work between peers. We can also do it as a follow up PR.

Reply:
Yeah, let's do that as a follow up.